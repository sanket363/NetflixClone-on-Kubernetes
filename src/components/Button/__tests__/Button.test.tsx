import {
  Primary,
  Secondary,
  IconRound,
  PrimaryWithIcon,
  SecondaryWithIcon,
  Alternate,
  IconRoundSecondary,
  PrimaryWithCustomClass,
} from './Button.stories'
import { render } from '@testing-library/react'

describe('Button constructor', (): void => {
  test('should render primary button', (): void => {
    const { container } = render(<Primary {...Primary.args} />)
    expect(container).toMatchSnapshot()
  })

  test('should render primary button with icon', (): void => {
    const { container } = render(<PrimaryWithIcon {...PrimaryWithIcon.args} />)
    expect(container).toMatchSnapshot()
  })

  test('should render secondary button', (): void => {
    const { container } = render(<Secondary {...Secondary.args} />)
    expect(container).toMatchSnapshot()
  })

  test('should render secondary button with icon', (): void => {
    const { container } = render(
      <SecondaryWithIcon {...SecondaryWithIcon.args} />
    )
    expect(container).toMatchSnapshot()
  })

  test('should render icon round button', (): void => {
    const { container } = render(<IconRound {...IconRound.args} />)
    expect(container).toMatchSnapshot()
  })

  test('should render secondary icon round button', (): void => {
    const { container } = render(
      <IconRoundSecondary {...IconRoundSecondary.args} />
    )
    expect(container).toMatchSnapshot()
  })

  test('should render alternate button', (): void => {
    const { container } = render(<Alternate {...Alternate.args} />)
    expect(container).toMatchSnapshot()
  })
  test('should render custom css class name', (): void => {
    const { container } = render(<Primary {...PrimaryWithCustomClass.args} />)
    expect(container).toMatchSnapshot()
  })
})
