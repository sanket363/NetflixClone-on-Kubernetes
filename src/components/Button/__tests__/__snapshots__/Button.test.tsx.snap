// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button constructor should render alternate button 1`] = `
<div>
  <button
    class="n-button with-label Alternate"
  >
    <span>
      Alternate
    </span>
  </button>
</div>
`;

exports[`Button constructor should render custom css class name 1`] = `
<div>
  <button
    class="n-button with-label custom-class-name Primary"
  >
    <span>
      Primary Custom
    </span>
  </button>
</div>
`;

exports[`Button constructor should render icon round button 1`] = `
<div>
  <button
    class="n-button with-icon IconRound"
  >
    <play-button.svg />
  </button>
</div>
`;

exports[`Button constructor should render primary button 1`] = `
<div>
  <button
    class="n-button with-label Primary"
  >
    <span>
      Primary
    </span>
  </button>
</div>
`;

exports[`Button constructor should render primary button with icon 1`] = `
<div>
  <button
    class="n-button with-icon with-label Primary"
  >
    <play-button.svg />
    <span>
      Primary Ic
    </span>
  </button>
</div>
`;

exports[`Button constructor should render secondary button 1`] = `
<div>
  <button
    class="n-button with-label Secondary"
  >
    <span>
      Secondary
    </span>
  </button>
</div>
`;

exports[`Button constructor should render secondary button with icon 1`] = `
<div>
  <button
    class="n-button with-icon with-label Secondary"
  >
    <play-button.svg />
    <span>
      Secondary Ic
    </span>
  </button>
</div>
`;

exports[`Button constructor should render secondary icon round button 1`] = `
<div>
  <button
    class="n-button with-icon IconRoundSecondary"
  >
    <play-button.svg />
  </button>
</div>
`;
